docs/development/url_previews.md

```diff
@@ -34,15 +34,17 @@ When Synapse is asked to preview a URL it does the following:
       2. Generates an Open Graph response based on image properties.
    5. If the media is HTML:
       1. Decodes the HTML via the stored file.
-      2. Generates an Open Graph response from the HTML.
-      3. If an image exists in the Open Graph response:
-         1. Downloads the URL and stores it into a file via the media storage
-            provider and saves the local media metadata.
-         2. Generates thumbnails.
-         3. Updates the Open Graph response based on image properties.
-   6. If the media is JSON and an oEmbed URL was found:
```

Comment:
This part is still valid if the oEmbed is downloaded via direct configuration instead of auto-discovery.

Reply:
Oh I see, this code is still not completely clear to me, but I'll try a fix

Reply:
Yeah, it is a bit confusing (which is why I tried to document it!) It seems that I failed to update it when I added the auto-discovery though.

I wonder if a flow-chart would be clearer...

Please feel free to reach out in #synapse-dev:matrix.org if you have questions! I'm more than happy to help!